File: lib/src/test/java/com/snap/ui/seeking/ReversingSeekableTest.java
package com.snap.ui.seeking;

import java.util.List;

import com.google.common.collect.Lists;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ReversingSeekableTest {

    Seekable<Integer> seekable;

    @Before
    public void setup() {
        seekable = new ListSeekable<>(Lists.newArrayList(1, 2, 3));
    }

    @Test
    public void basic_test() throws Exception {
        Seekable<Integer> reversed = Seekables.reverse(seekable);
        List<Integer> list = Lists.newArrayList(reversed);
        assertEquals(Lists.newArrayList(3, 2, 1), list);

        Seekable<Integer> forward = Seekables.reverse(list);
        List<Integer> list2 = Lists.newArrayList(forward);
        assertEquals(Lists.newArrayList(1, 2, 3), list2);
    }
}

File: lib/src/test/java/com/snap/ui/seeking/ReversingSeekableTest.kt
package com.snap.ui.seeking

import com.snap.ui.seeking.Seekables.reverse
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class ReversingSeekableTest {

    private lateinit var seekable: Seekable<Int>

    @Before
    fun setup() {
        seekable = ListSeekable(listOf(1, 2, 3))
    }

    @Test
    @Throws(Exception::class)
    fun basic_test() {
        val reversed = reverse(seekable)
        val list = reversed.toList()
        Assert.assertEquals(listOf(3, 2, 1), list)

        val forward = reverse(list)
        val list2: List<Int> = forward.toList()
        Assert.assertEquals(listOf(1, 2, 3), list2)
    }
}
